tardis/util.py

```diff
@@ -52,121 +52,114 @@ def __str__(self):
-   >>> ints = [roman_to_int(n) for n in nums]
-   >>> print r == ints
-   1
-
-   >>> roman_to_int('VVVIV')
-   Traceback (most recent call last):
-    ...
-   ValueError: input is not a valid roman numeral: VVVIV
-   >>> roman_to_int(1)
-   Traceback (most recent call last):
-    ...
-   TypeError: expected string, got <type 'int'>
-   >>> roman_to_int('a')
-   Traceback (most recent call last):
-    ...
-   ValueError: input is not a valid roman numeral: A
-   >>> roman_to_int('IL')
-   Traceback (most recent call last):
+
+def int_to_roman(int_input):
+    """
+    from http://code.activestate.com/recipes/81611-roman-numerals/
+    Convert an integer to Roman numerals.
+
+    :param int_input: an integer between 1 and 3999
+    :returns result: roman equivalent string of passed :param{int_input}
+
+    Examples:
+    >>> int_to_roman(0)
+    Traceback (most recent call last):
+    ValueError: Argument must be between 1 and 3999
+
+    >>> int_to_roman(-1)
+    Traceback (most recent call last):
+    ValueError: Argument must be between 1 and 3999
+
+    >>> int_to_roman(1.5)
+    Traceback (most recent call last):
+    TypeError: expected integer, got <type 'float'>
+
+    >>> for i in range(1, 21): print int_to_roman(i),
     ...
-   ValueError: input is not a valid roman numeral: IL
-   """
-   if type(input) != type(""):
-      raise TypeError, "expected string, got %s" % type(input)
-   input = input.upper()
-   nums = ['M', 'D', 'C', 'L', 'X', 'V', 'I']
-   ints = [1000, 500, 100, 50,  10,  5,   1]
-   places = []
-   for c in input:
-      if not c in nums:
-         raise ValueError, "input is not a valid roman numeral: %s" % input
-   for i in range(len(input)):
-      c = input[i]
-      value = ints[nums.index(c)]
-      # If the next place holds a larger number, this value is negative.
-      try:
-         nextvalue = ints[nums.index(input[i +1])]
-         if nextvalue > value:
-            value *= -1
-      except IndexError:
-         # there is no next place.
-         pass
-      places.append(value)
-   sum = 0
-   for n in places: sum += n
-   # Easiest test for validity...
-   if int_to_roman(sum) == input:
-      return sum
-   else:
-      raise ValueError, 'input is not a valid roman numeral: %s' % input
+    I II III IV V VI VII VIII IX X XI XII XIII XIV XV XVI XVII XVIII XIX XX
+    >>> print int_to_roman(2000)
+    MM
+    >>> print int_to_roman(1999)
+    MCMXCIX
+    """
+    if not isinstance(int_input, int):
+        raise TypeError("Expected integer, got %s" % type(int_input))
+    if not 0 < int_input < 4000:
+        raise ValueError("Argument must be between 1 and 3999")
+
+    int_roman_tuples = [(1000, 'M'), (900, 'CM'), (500, 'D'), (400, 'CD'),
+                        (100 , 'C'), (90 , 'XC'), (50 , 'L'), (40 , 'XL'),
+                        (10  , 'X'), (9  , 'IX'), (5  , 'V'), (4  , 'IV'), (1, 'I')]
+
+    result = ''
+    for iterable, int_roman_tuple in enumerate(int_roman_tuples):
```

Comment:
I don't think you need enumerate here. There must be a way to iterate and get int and str directly from the loop.
```
for int, str in int_roman_tuples:
```
Something like that but working(haven't tested only guessed that it won't that easily).